### DISCUSSION TITLE
Dynamic Datasource Custom HTTP Headers

### DISCUSSION THREAD

**Discussion by rhysxevans** [grafana/grafana#68039] [Data Sources]
Hi

Firstly apologies if I am asking a dumb question, or I have missed some documentation, but I would like the ability to set the data sources Custom HTTP Headers dynamically.

My use case is, via a OAUTH login into Grafana, Grafana is aware of my username and some other values that can be passed through in for example the X-ID-token. But lets just focus on the username and role for now.

I would like to setup the Custom HTTP Headers as follows

```
user: ${logged_in_username}
role: ${logged_in_users_role}
```

Is this possible ? If so what data is available to use for these substitutions ?

Any help is appreciated

Thanks

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82688. Feel free to continue the discussion around this there. Thank you!
